components/map/index.jsx

```diff
@@ -0,0 +1,162 @@
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+	/**
+	 * HTML id for component.
+	 */
+	id: PropTypes.string,
+	/**
+	 * Whether the modal containing the map is open. Only for `modal` variant
+	 */
+	isModalOpen: PropTypes.bool,
+	/**
+	 * Title for the Map component.
+	 */
+	title: PropTypes.string,
+	/**
+	 * Accepts a single Map location
+	 */
+	location: PropTypes.string,
+	/**
+	 * Accepts Map locations as an array of Objects
+	 */
+	locations: PropTypes.arrayOf(PropTypes.object),
+	/**
+	 * Accepts a Google Map API Key that will be used for showing the map
+	 */
+	googleAPIKey: PropTypes.string.isRequired,
+	/**
+	 * Accepts a node that will be shown at the footer of the modal containing the map. Only for `modal` variant
+	 */
+	modalFooter: PropTypes.node,
+	/**
+	 * Selects the variant of Map component
+	 */
+	variant: PropTypes.oneOf(['standalone', 'modal']),
+};
+
+const defaultProps = {
+	isModalOpen: true,
+	variant: 'standalone',
+};
+
+/**
+ * A map component is used to find a location
+ */
+class Map extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			selected: 0,
+		};
+		this.generatedId = shortid.generate();
+	}
+
+	/**
+	 * Get the Map's HTML id. Generate a new one if no ID present.
+	 */
+	getId() {
+		return this.props.id || this.generatedId;
+	}
+
+	render() {
+		const mapContainer = (
+			<div
+				id={this.getId()}
+				className={classNames(
+					`slds-grid`,
+					{'slds-has-coordinates' : this.props.locations},
+					this.props.className
+				)}
+			>
+				<div className="slds-map_container">
+					<div className="slds-map">
+						<iframe
+							id={`${this.getId()}-google-map`}
+							title="Google Maps iframe"
+							src={`https://www.google.com/maps/embed/v1/place?key=${
+								this.props.googleAPIKey
+							}&q=${ this.props.locations ?
+								encodeURIComponent(this.props.locations[this.state.selected].address)
+								: encodeURIComponent(this.props.location)
+							}`}
+						/>
+					</div>
+				</div>
+				{ this.props.locations ? (
+				<div className="slds-coordinates">
+					<div className="slds-coordinates__header">
+						<h2 className="slds-coordinates__title">{`${this.props.title} (${
+							this.props.locations.length
+						})`}</h2>
+					</div>
+					<ul className="slds-coordinates__list">
+						{this.props.locations.map((location, i) => (
+							<li key={location.name} className="slds-coordinates__item">
+								<span className="slds-assistive-text" aria-live="polite" />
+								<button
+									type="button"
+									onClick={() => this.setState({ selected: i })}
+									className="slds-coordinates__item-action slds-button_reset slds-media"
+									aria-pressed={this.state.selected === i}
+								>
+									<span className="slds-media__figure">
+										<Icon category="standard" name="account" />
```

Comment:
There's arren't always going to be `accounts`. Objects in `locations` colleciton should have a shape similar to `combobox` options

```
	 * * `icon`: An `Icon` component. (not used in read-only variant)
	 * * `id`: A unique identifier string.
	 * * `label`: A primary string of text for a menu item or group separator.
	 * * `subTitle`: A secondary string of text added for clarity. (optional)
```
https://github.com/salesforce/design-system-react/blob/master/components/combobox/combobox.jsx#L215